app/static/src/app/components/options/NumericInput.vue

```diff
@@ -83,22 +81,24 @@ export default defineComponent({
             const cleanedValue = newVal.replace(/,/g, "");
             const numeric = parseFloat(cleanedValue);
             if (!Number.isNaN(numeric)) {
-                // min max validation
-                if (numeric <= props.max && numeric >= props.min) {
-                    validatedNumeric.value = numeric;
+                let validatedNumeric: number;
+
+                // min/max validation
+                if (numeric > props.maxAllowed) {
+                    validatedNumeric = props.maxAllowed;
+                    errorTooltipProps.content = `Please enter a value no greater than ${props.maxAllowed}`;
+                } else if (numeric < props.minAllowed) {
+                    validatedNumeric = props.minAllowed;
+                    errorTooltipProps.content = (props.minAllowed === 0)
+                        ? "Please enter a non-negative number"
+                        : `Please enter a value no less than ${props.minAllowed}`;
```

Comment:
I'm not sure you need this special case - "Please enter a value no less than 0" seems just as useful as "Please enter a non-negative number"

Reply:
I thought it would be a nice touch as 0 is special. It feels more human to say non-negative number than no less than 0.

Reply:
Lol, I actually thought 'non-negative' sounded a bit too techie! But I don't think it matters too much!